File: RedSheet.DbEntities/Models/vApplicationModule.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vApplicationModules", Schema = "dbo")]
	[Serializable]
	public partial class vApplicationModule  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ApplicationModuleId { get; set; } 
 
	[Required]
	public bool IsRoot { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ModuleMasterId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string ModuleMasterName { get; set; } 

	public Nullable<int> ParentApplicationModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1)]
	public string VisibleActionItem { get; set; } 

	public vApplicationModule()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-application-module.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vApplicationModule {
    constructor(vApplicationModule?: vApplicationModule )  {
        let properties = [ "applicationModuleId", "isRoot", "moduleMasterId", "moduleMasterName", "parentApplicationModuleId", "visibleActionItem",];
        for (let property of properties)
            if (vApplicationModule && vApplicationModule[property])
                this[property] = vApplicationModule[property];
    }
 
	applicationModuleId : number =   0 ;
 
	isRoot : boolean = false ;
 
    @range(1,2147483647)
	moduleMasterId : number =   undefined;
 
    @required()
    @maxLength(100)
	moduleMasterName : string =   undefined;
 
	parentApplicationModuleId : number =   undefined;
 
    @required()
    @maxLength(1)
	visibleActionItem : string =   undefined;


}
